### ISSUE TITLE
ArrayFire Exception (Internal error:998)

### ISSUE THREAD

**Issue by deejayfrankcesco** [arrayfire/arrayfire#1631]
I tried to compile the example (on macOS Sierra 10.12): graphics -> plot3 and It return signal SIGABRT in runtime. The log in command prompt is:

---

```
ArrayFire v3.4.1 (CPU, 64-bit Mac OSX, build b9055b1)
[0] Intel: Intel(R) Xeon(R) CPU E5-1650 0 @ 3.20GHz, 16384 MB, Max threads(12) 
2016-10-19 18:12:49.314745 AFtest[1696:222822] [General] ERROR: Setting <private> as the first responder for window <private>, but it is in a different window ((null))! This would eventually crash when the view is freed. The first responder will be set to nil.
<private>
InfoLog:
WARNING: Output of vertex shader 'hpoint' not read by fragment shader

ArrayFire Exception (Internal error:998):
Forge Exception (OpenGL Error:2001):
In function void attachAndLinkProgram(GLuint, Shaders)
In file src/backend/opengl/common.cpp:102
OpenGL Program Compilation Failed
In function forge::Plot::Plot(const unsigned int, const dtype, const ChartType, const PlotType, const MarkerType)
In file src/api/cpp/plot.cpp:23

In function void af::Window::plot(const af::array &, const af::array &, const af::array &, const char *const)
In file src/api/cpp/graphics.cpp:95
libc++abi.dylib: terminating with uncaught exception of type af::exception: ArrayFire Exception (Internal error:998):
Forge Exception (OpenGL Error:2001):
In function void attachAndLinkProgram(GLuint, Shaders)
In file src/backend/opengl/common.cpp:102
OpenGL Program Compilation Failed
In function forge::Plot::Plot(const unsigned int, const dtype, const ChartType, const PlotType, const MarkerType)
In file src/api/cpp/plot.cpp:23

In function void af::Window::plot(const af::array &, const af::array &, const af::array &, const char *const)
In file src/api/cpp/graphics.cpp:95
```

---

I noticed that this error is present in all 3D plotting example.
What should I do?

**Comment by pavanky** [comment]
Are you running this remotely?

**Comment by deejayfrankcesco** [comment]
No, on local machine

**Comment by shehzan10** [comment]
Am I able to reproduce this error. Will look into it.

**Comment by shehzan10** [comment]
This issue is originating from Forge. I opened an issue there https://github.com/arrayfire/forge/issues/118.
This fix will be included in v3.4.2 of ArrayFire

**Comment by shehzan10** [comment]
This issue has been fixed in [Forge](https://github.com/arrayfire/forge). If you would like to have this fixed immediately at your end, the best solution is to clone and compile Forge devel branch from source and simple replace the existing `libforge*.dylib` files in `/usr/local/lib`.

**Comment by deejayfrankcesco** [comment]
Many thanks, you are the best! 
I’ll try soon!

Regards

**Comment by deejayfrankcesco** [comment]
Perfect, It works.
When will the fixed build release?

Regards

**Comment by shehzan10** [comment]
Fixed in hotfix-3.4.2. Release will follow.
